jobs/akkul_cards_gold.py

```diff
@@ -0,0 +1,88 @@
+"""
+This script will fetch all the cards from silver unique cards table and 
+- filter the table for specific user
+- filter data for specific timeframe passed in the script
+- save it to user Delta Lake Gold table
+"""
+
+import os
+import sys
+from datetime import datetime, timedelta
+import argparse
+from loguru import logger
+from pyspark.sql.functions import col, from_utc_timestamp, to_date
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from helpers import spark_helper as sh, common_functions as cf, trello_functions as tf
+from config import delta_table_conf as dc
+
+def get_user_cards(spark, start_date, end_date):
+    """
+    Fetch all the cards data for the given user within the given date range 
+    and save it into a Delta Lake Gold table
+    :param spark: Spark session
+    :param start_date: start date
+    :param end_date: end date
+    :return none
+    """
+    path = dc.akkul_cards_gold_path
+    try:
+        # Fetch the cards data from the refined cards silver delta table
+        refined_cards_data = spark.read.format(
+            "delta").load(dc.cards_unique_silver_table_path)
+        logger.info(
+            f'Fetched refined card data from silver delta table')      
+
+        # Convert from_date to a datetime object
+        start_date = datetime.strptime(start_date, '%Y-%m-%d')
+
+        # Convert "dateLastActivity" column to date format
+        refined_cards_data = refined_cards_data.withColumn("dateLastActivity", to_date(from_utc_timestamp("dateLastActivity", "UTC")))
+
+        # Filter the DataFrame for the given user and date range
+        filtered_cards_data = refined_cards_data.filter((col("card_members").like("%Akkul DN%")) & (col("dateLastActivity") >= start_date) & (col("dateLastActivity") <= end_date))
```

Comment:
Were you able to fetch any data? When I ran your script, the output showed 0 records. I looked at the data and saw your name is present as akkul.dn instead of Akkul DN. So, in this line when I changed to like(%akkul%), I was able to see many records.

Reply:
I initially though we had to search with our Trello display name. I didn't see any records in the parquet file. When I tried with 'akkul' im able to see some data in the output. Have updated that as well